xrpl/asyncio/transaction/main.py

```diff
@@ -142,6 +144,51 @@ async def submit_transaction(
     raise XRPLRequestFailureException(response.result)
 
 
+async def multisign(transaction: Transaction, signers: List[Wallet]) -> Transaction:
+    """
+    Multisigns a transaction.
+
+    Args:
+        transaction: the transaction to be signed.
+        signers: the wallets with which to sign the transaction.
+
+    Returns:
+        The multisigned transaction.
+    """
+    tx_json = transaction.to_xrpl()
+    # A blob is a binary string representation of the transaction that is
+    # in hex format. We need to encode it into a blob to make it smaller
+    # for rippled to process.
+    tx_blobs = [
+        sign(
+            bytes.fromhex(
+                encode_for_multisigning(
+                    tx_json,
+                    signer.classic_address,
+                )
+            ),
+            signer.private_key,
+        )
+        for signer in signers
+    ]
+
+    # Individually signing the transaction with each signer.
+    # This would normally be handled by reaching out to the signing
+    # account owners in your application since you would not normally
+    # be in control of all the keys.
+    tx_dict = transaction.to_dict()
+    tx_dict["signers"] = [
+        Signer(
+            account=signers[idx].classic_address,
+            txn_signature=tx_blobs[idx],
+            signing_pub_key=signers[idx].public_key,
+        )
+        for idx in range(len(signers))
+    ]
+
+    return AccountSet.from_dict(tx_dict)
```

Comment:
I think this helper is doing a bit too much - like the comment says, you normally don't have access to the private keys in order to sign the transactions for yourself. Instead I think this helper should take individually signed transactions and combine them into a single transaction with all signatures. (Otherwise it doesn't generalize well)

The rough idea is basically that the individually signed transactions each have a field where each account's signature will end up after they sign, and those can be combined into the eventual "multisigned transaction"'s `Signers` field. Happy to chat in more detail about this.